### DISCUSSION TITLE
Blazor server how to set user who is logged in

### DISCUSSION THREAD

**Discussion by DennyFiguerres** [IvanJosipovic/BlazorApplicationInsights#242] [General]
in the instructions there is a section that uses a RemoteAuthenticatorView component and sets the user 
in blazor server using azure ad i have not been able to find a way to impliment that.

does anyone have a way to do this ?

**Comment by DennyFiguerres** [comment]
ok i *think* i got some code in but i still do not see the user in application insights.....
blazor server .

i have code that does AppInsights.SetAuthenticatedUserContext(user.FindFirst("preferred_username")?.Value); and in debug i see the user id for that call.... but in the azure application insights users its <undefined> and i do not knopw what else to check to find it or why its not working....

**Comment by bdparkhurst** [comment]
@DennyFiguerres I am also experiencing the same problem where I can't seem to get the user context to flow through at all.  I'm wondering if this code is setting the authenticated user too late after the telemetry is initialized.  It feels like no one is answering any questions on this site and I'm concerned to even use this library anymore.
